app/services/users.js

```diff
@@ -0,0 +1,25 @@
+const logger = require('../logger');
+const { databaseError, duplicatedRegisterError } = require('../errors');
+const { User } = require('../models');
+const { GET_USER_BY_MAIL_ERROR, USER_ALREADY_EXIST } = require('../../config/constants');
+
+const findUserByEmail = async mail => {
+  try {
+    const user = await User.findOne({ where: { mail } });
+    return user;
+  } catch (err) {
+    logger.error(databaseError(err.errors));
+    throw databaseError(GET_USER_BY_MAIL_ERROR);
+  }
+};
+
+exports.createUser = async user => {
+  try {
+    const existUser = await findUserByEmail(user.mail);
```

Comment:
For good practices we do not make the call from a service to another service, this validation could be done from the middlewares and so the request can be cut from much earlier if necessary.